### ISSUE TITLE
Chatbot: Research on how to fix the chat memory (please add more details to this point)

### ISSUE THREAD

**Issue by SabaO7** [SWRIL-AI-NLP-Translation/swril-chatbot#9]

**Comment by kaiden-liu** [comment]
https://api.python.langchain.com/en/latest/memory/langchain.memory.chat_message_histories.redis.RedisChatMessageHistory.html

**Comment by kaiden-liu** [comment]
oops

**Comment by VionaWang** [comment]
In short, we need a session for each user to avoid mixing up conversations with each other. After researching for a bit, one way of doing it is to link a session id and use a database to store the messages. The link Kaiden added above would have the details on the usage and how to coordinate with the framework.

**Comment by SabaO7** [comment]
but you mentioned you will be researching alternative methods, right?

**Comment by VionaWang** [comment]
Yes, we are currently looking into other solutions. The above are just some details that might be useful for Kent to implement the redis option into the framework.
